File: pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;
import java.util.Map;

public interface AddressService {
    List<Address> findListByLoginUser(String name);

    List<Provinces> selectProvincesList();

    List<Cities> selectCityList(String provinceid);

    List<Areas> selectTownList(String cityid);

    void save(Address address);

    void setdefault(String id, String name);

    void deleteAddress(Address address);
}

File: pinyougou-web-user/src/main/webapp/js/service/addressService.js
app.service('addressService',function($http){

    this.selectProvincesList=function(){
        return $http.get('../address/selectProvincesList.do');
    };

    this.selectCityList=function(provinceid){
        return $http.get('../address/selectCityList.do?provinceid='+provinceid);
    };

    this.selectTownList=function(cityid){
        return $http.get('../address/selectTownList.do?cityid='+cityid);
    };
    //增加
    this.save=function(address){
        return  $http.post('../address/save.do',address );
    };
    //修改默认地址
    this.setDefaultAddress=function(aid){
        return  $http.get('../address/setDefaultAddress.do?id='+aid );
    };
//获取当前登录账号的收货地址
    this.findAddressList=function(){
        return $http.get('address/findListByLoginUser.do');
    };
    this.deleteAddress=function(address){
        return $http.post('address/deleteAddress.do',address);
    }

});